Content.Server/Chemistry/EntitySystems/RehydratableSystem.cs

```diff
@@ -1,47 +1,40 @@
-﻿using Content.Server.Chemistry.Components;
-using Content.Server.Popups;
-using Content.Shared.FixedPoint;
-using JetBrains.Annotations;
+using Content.Server.Chemistry.Components;
+using Content.Shared.Popups;
 
-namespace Content.Server.Chemistry.EntitySystems
+namespace Content.Server.Chemistry.EntitySystems;
+
+public sealed class RehydratableSystem : EntitySystem
 {
-    [UsedImplicitly]
-    public sealed class RehydratableSystem : EntitySystem
+    [Dependency] private readonly SharedPopupSystem _popups = default!;
+    [Dependency] private readonly SolutionContainerSystem _solutions = default!;
+
+    public override void Initialize()
     {
-        [Dependency] private readonly SolutionContainerSystem _solutionsSystem = default!;
-        public override void Initialize()
-        {
-            base.Initialize();
+        base.Initialize();
 
-            SubscribeLocalEvent<RehydratableComponent, SolutionChangedEvent>(OnSolutionChange);
-        }
+        SubscribeLocalEvent<RehydratableComponent, SolutionChangedEvent>(OnSolutionChange);
+    }
 
-        private void OnSolutionChange(EntityUid uid, RehydratableComponent component, SolutionChangedEvent args)
+    private void OnSolutionChange(EntityUid uid, RehydratableComponent comp, SolutionChangedEvent args)
+    {
+        if (_solutions.GetReagentQuantity(uid, comp.CatalystPrototype) > comp.CatalystAmount)
```

Comment:
this would behave incorrectly since then injecting *any* reagent will hydrate it since the catalyst will have 0, the minimum and if >= then it will hydrate